### DISCUSSION TITLE
Which tab has been selected by the user

### DISCUSSION THREAD

**Discussion by Petopp** [zauberzeug/nicegui#2543] [Q&A]
### Question

Hello everyone,

Does anyone have a tip for me on how I can determine the currently selected tab by the user? 
I would need to store the current tab in a parameter (app.storage.browser("Tab_Selected")) so that I can adjust the searches etc. later.

```py
with ui.tabs().classes('w-full') as tabs:
    one = ui.tab('Bot1')
    two = ui.tab('Bot2')

with ui.tab_panels(tabs, value=one).classes('w-full'):  
    with ui.tab_panel(one):      
        ui.chat_message('Hi, how can I help you?', name='Bot1', stamp='now')      
    with ui.tab_panel(two):
        ui.chat_message('Hello, how can I help you?', name='Bot2', stamp='now')
```

Thank you already

Peter

**Comment by falkoschindler** [comment]
Hi @Petopp,

The name of the currently selected tab is the `value` property of the `ui.tabs` or `ui.tab_panels` element. You can bind it to the user storage like this:

```py
@ui.page('/')
def page():
    with ui.tabs().classes('w-full') as tabs:
        one = ui.tab('Bot1')
        two = ui.tab('Bot2')

    with ui.tab_panels(tabs, value='Bot1').classes('w-full').bind_value(app.storage.user, 'tab'):
        with ui.tab_panel(one):
            ui.chat_message('Hi, how can I help you?', name='Bot1', stamp='now')
        with ui.tab_panel(two):
            ui.chat_message('Hello, how can I help you?', name='Bot2', stamp='now')
```

Note how I hard-coded the initial value `'Bot1'`, because there's currently no official way to get the name of a `ui.tab` element. This is being discussed in feature request https://github.com/zauberzeug/nicegui/discussions/2314.
